### PR TITLE
Ahmed-java script exercises 

### PR DESCRIPTION
**Pull request by ahmedahashim**

### CODE REVIEW

File: week-1/G-strings-methods/exercise.js [hunk 1]
```diff
@@ -1,3 +1,6 @@
 // Start by creating a variable `message`
 
-console.log(message);
+var name = "ahmed";
+var nameLength = name.length;
+
+console.log("my name is " + name + " and my name is 5 characters long");
```
Comment (ivinapontes) [inline week-1/G-strings-methods/exercise.js]: Hey Ahmed, I think you can use the variable _nameLength_ in the console in order to make nameLength dynamic. :smile:

Comment (ahmedahashim) [reply week-1/G-strings-methods/exercise.js]: Thanks ,ivinapontes

**Comment by ahmedahashim** [comment]
week-1/G-strings-methods/exercise.js
@@ -3,4 +3,6 @@
var name = "ahmed";
var nameLength = name.length;

console.log("my name is " + name + " and my name is 5 characters long");
console.log(
  "my name is " + name + " and my name is" + nameLength + "characters long"
);

File: week-1/L-functions-nested/exercise.js [hunk 2]
```diff
@@ -3,3 +3,28 @@ var mentor2 = "Irina";
 var mentor3 = "Mimi";
 var mentor4 = "Rob";
 var mentor5 = "Yohannes";
+//The percentage function
+function percentage(numberOfStudents, numberOfMentors) {
+  var total = numberOfStudents + numberOfMentors;
+  var percentageOfStudents = (numberOfStudents / total) * 100;
+  var percentageOfMentors = (numberOfMentors / total) * 100;
+  printPercentage(percentageOfStudents, percentageOfMentors);
```
Comment (mtbdata711) [inline week-1/L-functions-nested/exercise.js]: I really struggled finding a way to return two values on this task - I ended up returning an array -  and this solution is very good! The way in which the total is generated through adding the remaining variables is also very creative and clean.

Comment (aldeez988) [reply week-1/L-functions-nested/exercise.js]: nice work @ahmedahashim .

File: week-1-practise/level-1/1-syntax-error-1.js [hunk 3]
```diff
@@ -1,14 +1,13 @@
...
 
-
 // Answer these questions:
-// 1. How many parameters does the function take?
-// 2. What is the function name?
-// 3. Where is the function called? with what parameters?
\ No newline at end of file
+// 1. How many parameters does the function take? 3
+// 2. What is the function name? addNumbers
+// 3. Where is the function called? with what parameters?  line 7 parameters 1,2,3
```
Comment (martynamaron) [inline week-1-practise/level-1/1-syntax-error-1.js]: Nice! Well done :)

File: week-1-practise/level-1/10-function-output-3.js [hunk 4]
```diff
@@ -1,6 +1,7 @@
 function concatenate(firstWord, secondWord, thirdWord) {
-    // Write the body of this function to concatenate three words together
+  // Write the body of this function to concatenate three words together
+  return firstWord.concat(secondWord, thirdWord);
```
Comment (martynamaron) [inline week-1-practise/level-1/10-function-output-3.js]: Great use of string methods!
How could you do it without methods though? :)

File: week-1-practise/level-1/2-syntax-error-2.js [hunk 5]
```diff
@@ -1,8 +1,8 @@
...
 
-function addNumbers(a, b, c)
-return a + b + c;
-
+function addNumbers(a, b, c) {
+  return a + b + c;
+}
 
 var result = addNumbers(1, 3, 4);
 console.log(result); // Expected result 8 - Run the program and make sure it works
```
Comment (martynamaron) [inline week-1-practise/level-1/2-syntax-error-2.js]: 👏

File: week-1-practise/level-1/3-syntax-error-3.js [hunk 6]
```diff
@@ -1,7 +1,7 @@
 // There is a syntax error with this code. Fix it.
 
-function(a, b, c) {
-    return a + b + c;
+function addNumbers(a, b, c) {
+  return a + b + c;
```
Comment (martynamaron) [inline week-1-practise/level-1/3-syntax-error-3.js]: Yup!

File: week-1-practise/level-1/5-logic-error-2.js [hunk 7]
```diff
@@ -1,7 +1,7 @@
 // The syntax for this function is valid but it has an error, find it and fix it.
 
 function trimWord(word) {
-  return word.trim;
+  return word.trim();
```
Comment (martynamaron) [inline week-1-practise/level-1/5-logic-error-2.js]: Great, and why did you have to add the `()`?

File: week-1-practise/level-1/6-logic-error-3.js [hunk 8]
```diff
@@ -1,7 +1,7 @@
 // The syntax for this function is valid but it has an error, find it and fix it.
 
 function trim(word) {
-  return "word".trim();
+  return word.trim();
```
Comment (martynamaron) [inline week-1-practise/level-1/6-logic-error-3.js]: What is `word` in this instance then? :)

File: week-1-practise/level-1/8-function-output-1.js [hunk 9]
```diff
@@ -1,12 +1,12 @@
...
 // Answer these questions:
-// - What does this program do?
-// - How many parameters does the function take?
-// - What is the function name?
-// - Where is the function called? with what parameters?
\ No newline at end of file
+// - What does this program do? generates random number from 0 to lessthan 10
+// - How many parameters does the function take? 0
+// - What is the function name? getNumber
+// - Where is the function called? with what parameters? line 5  with no parameters
```
Comment (martynamaron) [inline week-1-practise/level-1/8-function-output-1.js]: Good one!

File: week-1-practise/level-1/9-function-output-2.js [hunk 10]
```diff
@@ -1,14 +1,14 @@
...
-// - What is the function name?
-// - Where is the function called? with what parameters?
-// - What does the function return?
-// - How can this function be improved?
\ No newline at end of file
+// - What does this program do? Run it and see, or Google some of the methods used ot understand them."concatenate two words together"
+// - How many parameters does the function take? 2 parameters
+// - What is the function name? s
+// - Where is the function called? with what parameters? line 5 with parameters "code","your"
+// - What does the function return? "concatenate string"
```
Comment (martynamaron) [inline week-1-practise/level-1/9-function-output-2.js]: I think a clearer description could be "a string formed of concatenating two of the function string parameters"

File: week-1-practise/level-1/9-function-output-2.js [hunk 11]
```diff
@@ -1,14 +1,14 @@
...
+// - How can this function be improved? give more descriptive name
```
Comment (martynamaron) [inline week-1-practise/level-1/9-function-output-2.js]: What would you call it?

File: week-1-practise/level-2/2-format-currency.js [hunk 12]
```diff
@@ -6,7 +6,10 @@
   Write a function that transforms numbers into the format £0.00
 */
 
-function formatCurrency() {}
+function formatCurrency(price) {
+  // currency formatting
```
Comment (martynamaron) [inline week-1-practise/level-2/2-format-currency.js]: Good result!
In terms of comments, it's good to remember that even though they might seem useful and descriptive, we should aim to make our code - that is, function and variable names - clear and descriptive enough to minimise the need for comments :) I think in this case the function name `formatCurrency` is meaningful enough

File: week-1-practise/level-2/3-currency-conversion.js [hunk 13]
```diff
@@ -5,7 +5,10 @@
   Write a function that converts a price to USD (exchange rate is 1.4 $ to £)
 */
 
-function convertToUSD() {}
+function convertToUSD(price) {
+  // conversion price
+  return price * 1.4;
```
Comment (martynamaron) [inline week-1-practise/level-2/3-currency-conversion.js]: Same thing about the comments, but great solution!

File: week-1-practise/level-2/4-currency-conversion-2.js [hunk 14]
```diff
@@ -7,8 +7,17 @@
...
+  return ((price + foreignTransactions(price)) * 1.4).toFixed(2); //price in USD
+}
+function convertToBRL(price) {
+  return ((price + foreignTransactions(price)) * 4.89).toFixed(2); // price in BRL
+}
+
+function foreignTransactions(price) {
+  // transaction fee
+  return price * 0.01;
+}
```
Comment (martynamaron) [inline week-1-practise/level-2/4-currency-conversion-2.js]: Well done!
Can you think of a way to simplify this code? For example, in the conversion functions, you always use the `price` variable and add the fee to it separately. Could be done in one function, no?

File: week-1-practise/level-3/1-piping.js [hunk 15]
```diff
@@ -11,7 +11,7 @@
 */
 
 var startingValue = 5;
-
+var result = format(add(multiply(startingValue, 2), 10)); // one line code
```
Comment (martynamaron) [inline week-1-practise/level-3/1-piping.js]: Nice! Can you explain which function is evaluated first, and which last?

File: week-1/G-strings-methods/exercise2.js [hunk 16]
```diff
@@ -1,3 +1,9 @@
-var name = " Daniel  ";
+var name = " Ahmed  ";
 
-console.log(message);
+console.log(
+  "My name is " +
+    name.trim() +
+    " and my name is " +
+    name.trim().length +
```
Comment (martynamaron) [inline week-1/G-strings-methods/exercise2.js]: That's a good one! What would happen if you didn't use `trim()`? :)

File: week-1/H-numbers/exercise.js [hunk 17]
```diff
@@ -1 +1,5 @@
 // Start by creating a variables `numberOfStudents` and `numberOfMentors`
+var numberOfStudents = 15;
+var numberOfMentors = 8;
+total = numberOfStudents + numberOfMentors;
```
Comment (martynamaron) [inline week-1/H-numbers/exercise.js]: It compiles and runs fine, but remember about initialising your variables correctly - what are you missing on line 4?
Also, rethink the name ;)

File: week-1/I-floats/exercise.js [hunk 18]
```diff
@@ -1,2 +1,7 @@
 var numberOfStudents = 15;
 var numberOfMentors = 8;
+var total = numberOfStudents + numberOfMentors;
```
Comment (martynamaron) [inline week-1/I-floats/exercise.js]: Again, minor thing, but renaming this variable might make it clearer what it represents :)

File: week-1/L-functions-nested/exercise.js (hunk 2 above)
Comment (martynamaron) [reply week-1/L-functions-nested/exercise.js]: This solution is quite creative, yes, but you don't need to calculate both at the same time! One job of a developer is to look for improving code and adhering do the DRY principle (do you know what it means?). In here - you're calculating the student percentage and then the mentors - but are you sure you have to? `numberOfStudents` is just a variable - you always obtain a percentage of something in the same way! You just need the total and the number of "things". So - how can this be improved then, without using an array either?

File: week-1/L-functions-nested/exercise.js [hunk 19]
```diff
@@ -3,3 +3,28 @@ var mentor2 = "Irina";
...
+  console.log("Percentage students: " + Math.round(percentageOfStudents) + "%");
+  console.log("Percentage mentors: " + Math.round(percentageOfMentors) + "%");
+}
+
+//prints the metors greeting
+function printGreeting(name) {
+  console.log("Hello " + name.toUpperCase());
+}
+
+percentage(15, 8);
```
Comment (martynamaron) [inline week-1/L-functions-nested/exercise.js]: After you have time to think about my previous comment, maybe you could address this function as well - `percentage` as a name doesn't really give away what it really does :)

**Comment by martynamaron** [comment]
Very well done! 👏 you've completed all the exercises and I've only left some minor comments :) Looks like you've really grasped the idea of variables and functions! 🎉 Make sure to be active in the channel and help other students who are struggling too :)

The comments I've left can be addressed as part of the next week's homework :)
Happy coding and happy holidays! ☃️

File: week-1-practise/level-1/1-syntax-error-1.js (hunk 3 above)
Comment (ahmedahashim) [reply week-1-practise/level-1/1-syntax-error-1.js]: Thanks

File: week-1-practise/level-1/5-logic-error-2.js (hunk 7 above)
Comment (ahmedahashim) [reply week-1-practise/level-1/5-logic-error-2.js]: trim() is a native code  in java script. Without added () to trim the result will be undefined.

File: week-1-practise/level-1/6-logic-error-3.js (hunk 8 above)
Comment (ahmedahashim) [reply week-1-practise/level-1/6-logic-error-3.js]: "code your future"

File: week-1-practise/level-1/10-function-output-3.js (hunk 4 above)
Comment (ahmedahashim) [reply week-1-practise/level-1/10-function-output-3.js]: function concatenate(firstWord, secondWord, thirdWord) {
  // Write the body of this function to concatenate three words together
  return firstWord + secondWord +t hirdWord;
}

File: week-1-practise/level-1/9-function-output-2.js (hunk 11 above)
Comment (ahmedahashim) [reply week-1-practise/level-1/9-function-output-2.js]: concatenateTwoString

File: week-1-practise/level-1/2-syntax-error-2.js (hunk 5 above)
Comment (ahmedahashim) [reply week-1-practise/level-1/2-syntax-error-2.js]: Thanks

File: week-1-practise/level-1/3-syntax-error-3.js (hunk 6 above)
Comment (ahmedahashim) [reply week-1-practise/level-1/3-syntax-error-3.js]: Thanks

File: week-1-practise/level-1/8-function-output-1.js (hunk 9 above)
Comment (ahmedahashim) [reply week-1-practise/level-1/8-function-output-1.js]: Thanks

File: week-1/K-functions-parameters/exercise4.js [hunk 20]
```diff
@@ -2,4 +2,8 @@
 
 // Call the function and assign to a variable `sum`
 
+function add(num1, num2) {
```
[the remark this answers is not in the document]
Comment (ahmedahashim) [reply week-1/K-functions-parameters/exercise4.js]: addTwoNumber

File: week-1/K-functions-parameters/exercise4.js [hunk 21]
```diff
@@ -2,4 +2,8 @@
...
+  return num1 + num2;
+}
+var sum = (13, 124);
```
[the remark this answers is not in the document]
Comment (ahmedahashim) [reply week-1/K-functions-parameters/exercise4.js]: Yes, I forgot to add function name in "var sum"
change done

File: week-1-practise/level-3/1-piping.js (hunk 15 above)
Comment (ahmedahashim) [reply week-1-practise/level-3/1-piping.js]: multiply function evaluated first .Add function evaluated last

File: week-1/H-numbers/exercise.js (hunk 17 above)
Comment (ahmedahashim) [reply week-1/H-numbers/exercise.js]: I missed to declare variable
variable name could be totalNumber
change done

File: week-1-practise/level-2/3-currency-conversion.js (hunk 13 above)
Comment (ahmedahashim) [reply week-1-practise/level-2/3-currency-conversion.js]: Can I say "format currency"

File: week-1-practise/level-2/2-format-currency.js (hunk 12 above)
Comment (ahmedahashim) [reply week-1-practise/level-2/2-format-currency.js]: Thanks  martynamaron .I got what do you mean

File: week-1-practise/level-2/4-currency-conversion-2.js (hunk 14 above)
Comment (ahmedahashim) [reply week-1-practise/level-2/4-currency-conversion-2.js]: Do you mean ,neglect foreignTransaction function?

File: week-1/L-functions-nested/exercise.js (hunk 19 above)
Comment (ahmedahashim) [reply week-1/L-functions-nested/exercise.js]: Thank you for your kind comments. You have given me a lot to work with.

**Pull request closed without merging**